bigquery_etl/dryrun.py

```diff
@@ -245,18 +302,61 @@ def is_valid(self):
             # we expect CREATE VIEW and CREATE TABLE to throw specific
             # exceptions.
             print(f"{self.sqlfile:59} OK")
+        elif (
+            error
+            and error.get("code", None) in [400, 403]
+            and "without a filter over column(s)" in error.get("message", "")
+        ):
+            # Some queries require a date filter
+            # (submission_date, submission_timestamp, etc.)
+            # We mark these requests as valid and add a date filter
+            # in get_referenced_table()
+            print(f"{self.sqlfile:59} OK")
+        elif (
+            error
+            and error.get("code", None) in [400, 403]
+            and "Syntax error: Expected end of input but got keyword WHERE"
+            in error.get("message", "")
+        ):
```

Comment:
I don't think we should be overriding the default dry-run behavior to return OK if there are legitimate errors. Maybe this should be conditional based on whether a flag is set during the constructor?